data/src/main/java/org/apache/iceberg/data/parquet/GenericParquetReaders.java

```diff
@@ -396,7 +409,12 @@ private FixedReader(ColumnDescriptor desc) {
     }
   }
 
-  static class RecordReader extends StructReader<Record, Record> {
+  public interface StructReaderFactory<T> {
+
+    StructReader<T, T> create(List<Type> types, List<ParquetValueReader<?>> readers, StructType struct);
+  }
+
+  static class RecordReader extends StructReader<GenericRecord, GenericRecord> {
```

Comment:
Why change this to use the `GenericRecord` instance rather than the `Record` interface?

I don't see much value in this change. We will just need to change it back if we want to add implementations of `Record` that are not generic, like we do with our internal classes that extend Avro's `IndexedRecord`. Ideally, I'd like to change those over to internal records eventually.

Reply:
I changed this because the `buildReader` method return the `GenericRecord` ParquetValueReader. Seems better to change the ParquetValueReader to be `Record` reader.